couchbase/src/main/scala/akka/stream/alpakka/couchbase/scaladsl/CouchbaseSession.scala

```diff
@@ -31,20 +31,15 @@ object CouchbaseSession {
    * the session is closed.
    */
   def apply(settings: CouchbaseSessionSettings,
-            bucketName: String)(implicit ec: ExecutionContext): Future[CouchbaseSession] = {
-    //wrap CouchbaseAsyncCluster.create up in the Future because it's blocking
-    val asyncCluster: Future[AsyncCluster] =
-      Future(settings.environment match {
-        case Some(environment) =>
-          CouchbaseAsyncCluster.create(environment, settings.nodes: _*)
-        case None =>
-          CouchbaseAsyncCluster.create(settings.nodes: _*)
-      }).map(_.authenticate(settings.username, settings.password))
+            bucketName: String)(implicit ec: ExecutionContext): Future[CouchbaseSession] =
+    openBucket(createClusterClient(settings), disconnectClusterOnClose = true, bucketName)
 
-    asyncCluster
-      .flatMap(c => RxUtilities.singleObservableToFuture(c.openBucket(bucketName), "").map((c, _)))
-      .map { case (c, bucket) => new CouchbaseSessionImpl(bucket, Some(c)) }
-  }
+  /**
+   * Create a session against the given bucket. The life-cycle of the `client` is the user's responsibility.
+   */
+  def apply(cluster: Future[AsyncCluster],
+            bucketName: String)(implicit ec: ExecutionContext): Future[CouchbaseSession] =
+    openBucket(cluster, disconnectClusterOnClose = false, bucketName)
```

Comment:
A bit weird taking a `Future`/`CompletionStage` as a param here, shouldn't it rather be `(AsyncCluster, String) => Future[CouchbaseSession]` and if a user has a `Future[AsyncCluster]` or `Future[String]` for that part it is their job to `map/flatMap` it.